GMX-Bots/gmx5-detect-frontruns/src/agent.ts

```diff
@@ -0,0 +1,205 @@
+                  victimTokenIn = tokenInVictim;
+                  victimTokenOut = tokenOutVictim;
+                  victimTokenInAmount = amountInVictim;
+                  victimTokenOutAmount = amountOutVictim;
+                  frontrunnerAddress = accountFront;
+                  frontrunnerProfit = amountOutBack - amountInFront;
+                  victimAddressAndTokens = addressAndTokens;
+                  frontrunnerAddressAndTokens = accountBack + token0 + token1;
+                }
+              }
+            });
+          }
+        }
+
+        if (!sandwichAttack) {
+          callHistory.set(accountBack + token0 + token1, [swapEvent, callHistoryIndex, txEvent.hash]);
+          callHistoryKeys[callHistoryIndex] = accountBack + token0 + token1;
+        }
+
+        callHistoryIndex++;
+        mapCleanerCounter++;
+
+        //callHistory cleaning
+        if (callHistoryIndex == CALL_HISTORY_SIZE) {
+          callHistoryIndex = 0;
+
+          if (mapCleanerCounter > 50000) {
+            mapCleanerCounter = 0;
+            Array.from(callHistory.keys()).forEach((addressAndTokens) => {
+              let beingUsed = false;
+              callHistoryKeys.forEach((key) => {
+                if (addressAndTokens == key) {
+                  beingUsed = true;
+                }
+              });
+              if (!beingUsed) {
+                callHistory.delete(addressAndTokens);
+              }
+            });
+          }
+        }
+
+        //if a gmx trade was frontrun with a sandwich attack, report it
+        if (sandwichAttack) {
+          callHistory.delete(victimAddressAndTokens);
+          callHistory.delete(frontrunnerAddressAndTokens);
+          findings.push(
+            Finding.fromObject({
+              name: "Sandwich Attack Frontrun",
+              description: `User ${victimAddress.toLowerCase()} suffered a sandwich attack`,
+              alertId: "GMX-05",
+              protocol: "GMX",
+              severity: FindingSeverity.Medium,
+              type: FindingType.Suspicious,
+              metadata: {
+                sandwichFrontTransaction: frontHash,
+                victimTransaction: victimHash,
+                sandwichBackTransaction: txEvent.hash,
+                victimAddress: victimAddress.toLowerCase(),
+                victimTokenIn: victimTokenIn.toLowerCase(),
+                victimTokenOut: victimTokenOut.toLowerCase(),
+                victimTokenInAmount: victimTokenInAmount.toString(),
+                victimTokenOutAmount: victimTokenOutAmount.toString(),
+                frontrunnerAddress: frontrunnerAddress.toLowerCase(),
+                frontrunnerProfit: frontrunnerProfit.toString(), //in terms of tokenIn
+              },
+            })
+          );
+        }
+      });
+    }
+    return findings;
+  };
+
+export default {
+  initialize: initialize(provider),
+  handleTransaction: provideHandleTx(GMX_ROUTER_ADDRESS, SWAP_EVENT, provider),
```

Comment:
this is not going to work, please do the following experiment:

1. Add a `console.log` in the `initialize` function to display the address you get from the `NetworkManager`.
2. Add a `console.log` in the line 50 for example to display `router`.
3. Do `npm run` on one of the supported networks and see what is printed.

Reach me on slack after that to discuss